src/printer.rs

```diff
@@ -31,32 +32,48 @@ impl<'a> Printer<'a> {
     }
 
     pub fn print_header(&mut self, filename: Option<&str>) -> Result<()> {
-        match self.options.style {
-            OptionsStyle::Full => {}
-            _ => return Ok(()),
+        if !self.options
+            .output_components
+            .contains(&OutputComponent::Header)
+        {
+            return Ok(());
         }
 
-        self.print_horizontal_line('┬')?;
-
-        write!(
-            self.handle,
-            "{}{} ",
-            " ".repeat(PANEL_WIDTH),
-            self.colors.grid.paint("│"),
-        )?;
-
-        writeln!(
-            self.handle,
-            "{}{}",
-            filename.map_or("", |_| "File: "),
-            self.colors.filename.paint(filename.unwrap_or("STDIN"))
-        )?;
-
-        self.print_horizontal_line('┼')
+        if self.options
+            .output_components
+            .contains(&OutputComponent::Grid)
+        {
+            self.print_horizontal_line('┬')?;
+
+            write!(
+                self.handle,
+                "{}{} ",
+                " ".repeat(PANEL_WIDTH),
+                self.colors.grid.paint("│"),
+            )?;
+
+            writeln!(
+                self.handle,
+                "{}{}",
+                filename.map_or("", |_| "File: "),
+                self.colors.filename.paint(filename.unwrap_or("STDIN"))
+            )?;
+
+            self.print_horizontal_line('┼')
+        } else {
+            writeln!(
```

Comment:
This is not quite the same behavior. probably best to avoid the duplication

Reply:
True, thanks for noticing. I'll leave this unchanged for the time being until we have a decision on whether `header` should have an effect without `grid`.